### ISSUE TITLE
Components package: deprecated lifecycle method warning in slot-fill

### ISSUE THREAD

**Issue by vindl** [WordPress/gutenberg#9017]
The components package is using the deprecated `componentWillUpdate` lifecycle method which produces console warnings when executed in React strict mode.

https://github.com/WordPress/gutenberg/blob/6928e41c8afd7daa3a709afdda7eee48218473b7/packages/components/src/slot-fill/fill.js#L25-L34

```
Warning: Unsafe lifecycle methods were found within a strict-mode tree:
    in Editor (created by _class)
    in _class (created by RemountOnPropChange(_class))
    in RemountOnPropChange(_class)
    ...

componentWillUpdate: Please update the following components to use componentDidUpdate instead: Fill
```

cc @youknowriad @gziolo

**Comment by youknowriad** [comment]
This one is probably not the easiest one to replace because I believe the "order" of which the fills get added is important. So I'm pinging @aduth he knows more about that.

**Comment by aduth** [comment]
It is important. I'll give it a look, and if there aren't tests already, will add them in the fix.

**Comment by aduth** [comment]
As I understand, this was fixed as of #15541.
